include/eve/detail/function/simd/x86/compress_store_impl.hpp

```diff
@@ -14,6 +14,74 @@
 #include <array>
 #include <bit>
 
+// sse2/sse3 ---------------------------------------------------------
+// switch based implementation
+
+namespace eve::detail
+{
+
+  template<relative_conditional_expr C, typename T, typename U>
+  EVE_FORCEINLINE
+  auto compress_store_impl_switch_4(C c,
+                                    wide<T, fixed<4>> v,
+                                    logical<wide<U, fixed<4>>> mask
+                                    )
+  {
+    auto [num, last_set] = compress_store_swizzle_mask_num[c](mask);
+    int count;
+
+    switch (num) {
+      case 0: { count = 0; v = v[pattern<3, 0, 0, 0>]; break; } // 000
```

Comment:
can we actually merge both the case and the comment by doing `case 0b001` and so on or is it risky ?